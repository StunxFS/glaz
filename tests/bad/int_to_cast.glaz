enum GameState {
	Intro,
	MainMenu,
	CityMap = 5,
	Credits = 9
}

fn main() {
	_ = cast(0, GameState); // OK
	_ = cast(1, GameState); // OK
	_ = cast(5, GameState); // OK
	_ = cast(9, GameState); // OK

	_ = cast(10, GameState); // BAD
}
