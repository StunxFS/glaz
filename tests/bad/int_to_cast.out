tests/bad/int_to_cast.glaz:14:12: error: 10 does not represent a valid value of enum `GameState`
     | 
  14 | 	_ = cast(10, GameState); // BAD
     | 	         ^^
glaz error: could not compile `int_to_cast`, aborting due to previous error.
